Remove wired-clients by interpreting RTM_DELNEIGH

christf opened the pull request:
Without this patch, wire-only clients accumulate over time and are never removed from l3roamd as shown by https://grafana.bremen.freifunk.net/d/000000002/node?orgId=1&var-node=5254002a92c8&refresh=1m&from=1559635941032&to=1560240681032&var-saveinterval=60

This patchset improves the build system somewhat and resolves the issue observed in the bremen network.

christf committed:
routemgr, wifistations: use log_debug and format output

christf committed:
cmake: Do not build dependencies twice

restructuring CmakeLists.txt to re-use compiled parts for both the l3roamd and
the l3roamd-test executables to speed up the build

christf committed:
Taskqueue: Allow to drop tasks

christf committed:
intercom: resolve initialization warning

replace struct client c = {0} with
struct client c = {}

christf committed:
clientmgr: re-format

christf committed:
clientmgr, routemgr: react on DELNEIGH

routemgr reacts on DELNEIGH netlink messages now, deleting the affected address
from the client within 2 minutes. When adding a new IP to a client, possibly
existing tasks are cancelled.

christf committed:
clientmgr: clientmgr_handle_info() use log_verbose replacing printf()

genofire commented:
Keep grafana client graph before we update it, after the test build:
![node](https://user-images.githubusercontent.com/6905586/59265262-4cc32400-8c45-11e9-9f25-294b752fe364.png)


Build is running: https://jenkins.bremen.freifunk.net/job/babel-test/147/

christf committed:
Allow setting the verbosity of the process via control socket

christf committed:
logging and comment cleanup

genofire commented:
Next Build is running: https://jenkins.bremen.freifunk.net/job/babel/148/

christf committed:
util: introduce STRBUFLEN to specify the length of the string buffer

This gets rid of the assumption that we are only printing strings with
INET6_ADDRSTRLEN and allows separate handling.

christf committed:
arp: make use of print_mac()

christf committed:
clientmgr: initialize oldclients vector, adjust output

christf committed:
icmp6.c: use strncmp instead of strcmp, adjust logging when wetting up interface

christf committed:
taskqueue: use data types from struct timespec

using time_t for the second-value instead hard-coded long or unsigned int
for nsec - using long instead unsigned int

christf committed:
taskqueue: run all tasks with past due dates in run()

this means that when overlapping timespecs are specified, the task is still
executed. On non-overlapping timespecs and long duration of tasks we are not
missing executions with this patch.

christf committed:
prefix.c: get rid of now useless debug message

checking if prefix contains address

christf committed:
clientmgr: make use of log_Verbose and print_ip and remove useless TODO

genofire commented:
Build is running: https://jenkins.bremen.freifunk.net/job/babel/154/

christf committed:
Remove CTX macro, use l3ctx structure everywhere

christf committed:
util: rework stringbuffer to allocate it in one or in multiple chunks

christf committed:
icmp6: icmp6_handle_ns_in() work around memory alignment issues when creating ns_task

christf committed:
print_client return char*, allow usage from printf/log_*

this re-uses the string-buffer from util.h

christf committed:
intercom: add_interface() avoid referencing empty ifname

christf committed:
prefix: parse_prefix() memory fixes

* initialize saveptr
* check result of strtok
* allocate tmp on the stack instead of the heap, getting rid of error label

christf committed:
routemgr: handle_msg() do not reference empty nh

christf committed:
taskqueue: run task immediately if due date is in the past

christf committed:
clarify exit error messages wrt options including unparsable parameter
